### PR TITLE
Upgrading dependencies + adding deprecation note in the README.

### PR DESCRIPTION
**Pull request by JPFrancoia**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,20 +10,20 @@ license = "MIT/Apache-2.0"
 name = "slack-hook2"
 readme = "README.md"
 repository = "https://github.com/clux/rust-slack"
-version = "0.10.0"
+version = "0.10.1"
```
Comment (JPFrancoia) [inline Cargo.toml:13]: We're just bumping dependencies, bumping patch version.

File: Cargo.toml [hunk 2]
```diff
@@ -10,20 +10,20 @@ license = "MIT/Apache-2.0"
...
+hex = "^0.4"
+serde = "^1"
+serde_derive = "^1"
+serde_json = "^1"
+url = { version = "^2", features = ["serde"] }
+thiserror = "^1.0"
 
 [dependencies.reqwest]
-version = "0.10.1"
+version = "~0.11.12"
```
Comment (JPFrancoia) [inline Cargo.toml:26]: reqwest tends to break from one version to another so less freedom in the version.

**Review by clux — APPROVED** [review]
LGTM. Thanks!

**Comment by clux** [comment]
0.10.1 published

**Pull request merged**